funnel/views/project.py

```diff
@@ -786,6 +786,16 @@ def settings(self):
     @render_with('project_comments.html.jinja2')
     @requires_roles({'reader'})
     def comments(self):
+        if current_auth.user.active_commentset_memberships is not None:
+            commentset_membership = (
+                current_auth.user.active_commentset_memberships.filter_by(
+                    commentset=self.obj.commentset
+                ).one_or_none()
+            )
+            if commentset_membership is not None:
+                commentset_membership.last_seen_at = db.func.utcnow()
+                db.session.commit()
+
```

Comment:
Can't add a subscription on just reading. There has to be some action. Also, what if `current_auth.user is None`?